modules/api/src/main/java/org/apache/ignite/table/criteria/CriteriaBuilder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.table.criteria;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents a criteria builder. Implementations of this interface are basic building blocks for performing criteria queries.
+ */
+public class CriteriaBuilder implements Criteria {
+    private final List<Criteria> lsCriteria = new ArrayList<>();
+    private final String columnName;
+
+    private CriteriaBuilder(String columnName) {
+        this.columnName = columnName;
+    }
+
+    public static CriteriaBuilder columnName(String columnName) {
```

Comment:
The builder pattern supposes `build()` method is called at last, when all the work is done, but there is no one.
Regarding this and a fact the object `implements Criteria`, I guess the builder is always in valid state... but looking at the implementation - I think it is not true.

What if a user creates a builder without calling `equals(..)` method
```
CriteriaBuilder.columnName('col1')
```
Will the builder be in valid state?

Reply:
Let's make Criteria and builder a separate classes/interfaces.

Reply:
I decided to remove `CriteriaBuilder` from this PR at all. We can add it later. It was optional